lib/surface/components/form/error_tag.ex

```diff
@@ -0,0 +1,124 @@
+defmodule Surface.Components.Form.ErrorTag do
+  @moduledoc """
+  A component inspired by `error_tag/3` that ships with `mix phx.new` in
+  `MyAppWeb.ErrorHelpers`.
+
+  Renders error messages if any exist regarding the given field.
+
+  ## Error Translation
+
+  Changeset errors are translated using the default error translator
+  that ships with phoenix when generated with `mix phx.new --no-gettext`.
+
+  When using Gettext, you can use configuration to route all errors through
+  the `MyAppWeb.ErrorHelpers.translate_error/1` function generated by Phoenix,
+  which utilizes `Gettext`. You need to provide a tuple with the module and
+  the name of the function (as an atom) as follows:
+
+  ```elixir
+  config :surface, :components, [
+    {Surface.Components.Form.ErrorTag, translator: {MyAppWeb.ErrorHelpers, :translate_error}}
+  ]
+  ```
+
+  There is also a `translator` prop which can be used on a case-by-case basis.
+  It overrides the configuration.
+
+  ## Examples
+
+  ```surface
+  <ErrorTag field="password" />
+  ```
+
+  ```surface
+  <Field name="password>
+    <ErrorTag />
+  </Field>
+  ```
+
+  ```surface
+  <ErrorTag phx_feedback_for="confirm_password_for_reset" />
+  ```
+
+  ```surface
+  <ErrorTag class="custom-css-classes" />
+  ```
+
+  ```surface
+  <ErrorTag translator={{ &CustomTranslationLib.translate_error/1 }} />
+  ```
+  """
+
+  use Surface.Component
+
+  import Phoenix.HTML.Form, only: [input_id: 2]
+
+  alias Surface.Components.Form.Input.InputContext
+
+  @doc "An identifier for the form"
+  prop form, :form
+
+  @doc "An identifier for the associated field"
+  prop field, :atom
+
+  @doc "Class or classes to apply to each error tag <span>"
+  prop class, :css_class
+
+  @doc """
+  A function that takes one argument `{msg, opts}` and returns
+  the translated error message as a string. If not provided, falls
+  back to Phoenix's default implementation.
+
+  This can also be set via config, for example:
+
+  ```elixir
+  config :surface, :components, [
+    {Surface.Components.Form.ErrorTag, translator: {MyApp.Gettext, :translate_error}}
+  ]
+  ```
+  """
+  prop translator, :fun
+
+  @doc """
+  If you changed the default ID on the input, provide it here.
+  (Useful when there are multiple forms on the same page, each
+  with an input of the same name. LiveView will exhibit buggy behavior
+  without assigning separate id's to each.)
+  """
+  prop phx_feedback_for, :string
+
+  def render(assigns) do
+    translate_error = assigns.translator || translator_from_config() || (&translate_error/1)
+
+    ~H"""
+    <InputContext assigns={{ assigns }} :let={{ form: form, field: field }}>
+      <span
+        :for={{ error <- Keyword.get_values(form.errors, field) }}
```

Comment:
What about the case we discussed about having `<Form for={{ :an_atom }}>`. It will crash with a `function :an_atom.errors/0 is undefined`. We have to handle that, right?

Reply:
So there was something about the [prior to this PR] implementation that caused a crash.

But in this case, even if you do `for={{ :atom }}`, we'll still have a form struct. I believe it's `form.data` that is nil in this case. But we're not referencing `form.data.errors`. We're referencing `form.errors`, which will be identical to `form.data.errors` as long as `form.data` is a changeset with `changeset.action != nil`. Otherwise `form.errors` will be `[]`. There is a test in this PR that demonstrates that it doesn't crash.